scripts/docs/parse_assets.py

```diff
@@ -31,34 +31,85 @@ def parse_assets(input_dirs: List[Path],
     """
     asset_count = 0
 
-    references = defaultdict(list)
+    references = {}
 
     for asset_config in util.find_assets(input_dirs, asset_config_filename, pattern=pattern):
         asset_count += 1
 
         asset_type, asset_name, asset_file_name, asset_description = \
             generate_asset_documentation.create_asset_doc(asset_config, asset_config.type)
 
-        references[asset_type].append((asset_name, asset_file_name, asset_description))
+        if asset_type not in references:
+            references[asset_type] = defaultdict(list)
+
+        for category in asset_config.categories:
+            references[asset_type][category].append((asset_name, asset_file_name, asset_description))
+
+        # If no categories are found, put the asset under "Uncategorized"
+        if len(asset_config.categories) == 0:
+            references[asset_type]["Uncategorized"].append((asset_name, asset_file_name, asset_description))
+
+        references[asset_type]["All"].append((asset_name, asset_file_name, asset_description))
 
     logger.print(f"{asset_count} asset(s) parsed")
 
     for asset_type in references:
+
+        category_docs_links_list = []
+
+        for category in references[asset_type]:
+            if category == "All":
+                continue
+
+            category_doc_file_name = f"{asset_type}s-{category.replace(' ', '-')}-documentation"
+            capitalized_category = category[0].upper() + category[1:]
```

Comment:
https://docs.python.org/3/library/stdtypes.html#str.title
```suggestion
            capitalized_category = category.title()
```

Reply:
For the category of "Responsible AI", .title() changes it to "Responsible Ai." Should we do some sort of combination or do you have any additional suggestions?